dsa/src/size.rs

```diff
@@ -0,0 +1,38 @@
+macro_rules! define_param_size {
+    ($l:literal, $n:literal) => {
+        ::paste::paste! {
+            #[doc = "DSA parameter size constant; L = " $l ", N = " $n]
+            pub const [<DSA_ $l _ $n>]: Self = Self::custom($l, $n);
+        }
+    };
+    (deprecated: $l:literal, $n:literal) => {
+        ::paste::paste! {
+            #[deprecated(note="This size constant has a security strength of under 112 bits per SP 800-57 Part 1 Rev. 5")]
+            #[doc = "DSA parameter size constant; L = " $l ", N = " $n]
+            pub const [<DSA_ $l _ $n>]: Self = Self::custom($l, $n);
+        }
+    };
+}
+
+/// DSA key size
+pub struct KeySize {
+    /// Bit size of p
+    pub(crate) l: u32,
+
+    /// Bit size of q
+    pub(crate) n: u32,
+}
+
+impl KeySize {
+    define_param_size!(deprecated: 1024, 160);
+    define_param_size!(2048, 224);
+    define_param_size!(2048, 256);
+    define_param_size!(3072, 256);
```

Comment:
It feels to me like these macros are just adding indirection and an additional dependency with `paste`.

Can't this simply be written as:

```suggestion
    /// DSA parameter size constant: L = 1024, N = 160
    #[deprecated(note = "This size constant has a security strength of under 112 bits per SP 800-57 Part 1 Rev. 5")]
    pub const DSA_1024_160: Self = Self { l: 1024, n: 160 };
    
    /// DSA parameter size constant: L = 2048, N = 224
    pub const DSA_2048_224: Self = Self { l: 2048, n: 224 };

    /// DSA parameter size constant: L = 2048, N = 256
    pub const DSA_2048_256: Self = Self { l: 2048, n: 256 };

    /// DSA parameter size constant: L = 3072, N = 256
    pub const DSA_3072_256: Self = Self { l: 3072, n: 256 };
```

Reply:
When I originally wrote the code I didn't like the idea of copy pasting a bunch of code and the dependency tree already had `paste` in it so I wrote this macro  

But here it's just an unnecessary dependency + indirection, I agree